File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        min_val = nums[0]
        total = nums[0]
        for n in nums[1:]:
            total += n
            min_val = min(min_val, total)
        return 1 - min(min_val, 0)
        
        

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.go
func minStartValue(nums []int) int {
    minVal := nums[0]
    total := nums[0]
    for _, n := range nums[1:] {
        total += n
        minVal = min(minVal, total)
    }
    return 1 - min(minVal, 0)
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}